### ISSUE TITLE
Add accessibility signal for when an edit has been accepted/applied

### ISSUE THREAD

**Issue by meganrogge** [microsoft/vscode#239373]
cc @jooyoungseo

When an individual edit has been accepted/applied, we need to indicate this with a signal.

cc @afre100

**Comment by AFre100** [comment]
@meganrogge zooming out, would this be an added sound to [this](https://github.com/microsoft/vscode/issues/230322) parallel conversation?

Would the full flow be this? 

-edit suggestion menu first pops up [New Edit Suggestion Alert Sound 1]
-User navigates through menu of 1+ edit suggestion(s) and triggers one�[new "edit selected" sound 3]
-An edit suggestion has been applied [New Edit Suggestion Applied Sound 2]

**Comment by meganrogge** [comment]
@AFre100 , different flow here. 

User navigates to an edit, uses the keybinding to apply the edit, we play the sound.

**Comment by AFre100** [comment]
@meganrogge @jrieken Apologies -- it's still unclear to me—how this audio cue isn't the same feedback as either ["edit selected" sound 3] or [New Edit Suggestion Applied Sound 2], as I proposed above. 

I'm wondering whether this a) a different category of Edits from Edit Suggestions; or maybe b) it's the same, but requires a different flow to get there, and there needs to be distinct sounds that communicate which flow someone is coming from. 

Can you please clarify? If you could provide a screengrab vid, that would help speed things up. Thanks! 

> [@meganrogge](https://github.com/meganrogge) zooming out, would this be an added sound to [this](https://github.com/microsoft/vscode/issues/230322) parallel conversation?
> 
> Would the full flow be this?
> 
> -edit suggestion menu first pops up [New Edit Suggestion Alert Sound 1] -User navigates through menu of 1+ edit suggestion(s) and triggers one�[new "edit selected" sound 3] -An edit suggestion has been applied [New Edit Suggestion Applied Sound 2]

**Comment by jrieken** [comment]
> User navigates to an edit, uses the keybinding to apply the edit, we play the sound.

Is this really something we do, like do we normally play audio cues to confirm an action happened that you just manually invoked? Like, me: "I pressed save", code "tata you pressed save"? We don't do the same for sighted users, like we don't show a notification for actions unless the outcome is surprising/failure

**Comment by meganrogge** [comment]
Yes we do - for example, the save audio cue. 

A sighted sees the change get applied. We need to communicate that to screen reader users.

**Comment by jrieken** [comment]
So, the rule is "when the UI changes, like no more dirty indications or no more change/review indication", then also play a sound?

**Comment by meganrogge** [comment]
This is case by case. If we played a sound for every interaction, it would be overwhelming. 

Confirming that edits have been applied is important because otherwise a user won't know that. 

Re the dirty indications etc, if a user wants that info, they can query the window.title with `activeEditorState`.

**Comment by jrieken** [comment]
> Confirming that edits have been applied is important because otherwise a user won't know that.

Just to be on the same page: Edits are already applied _before_ pressing accept. Editing just happens and accept just means "remove the review UI, e.g the red and green". It's being discussed but "Accept" should likely be called "OK"

**Comment by meganrogge** [comment]
Thanks for letting me know. Even so, the disappearing diff color signals something to sighted users. We also need to indicate this to screen reader users. 

Has this changed recently? I would expect that a user has a chance to review and accept each diff. I thought that was how it worked at one point, maybe I'm wrong?

**Comment by jrieken** [comment]
It was always like that, the only thing that has changed that we now even save before accept

**Comment by meganrogge** [comment]
Ok, how about we go with:

- accept (for sighted users, hides diff) play acceptance sound 
- discard (for sighted users, removes change) play discard sound

WDYT?

**Comment by jrieken** [comment]
Makes sense. Would we differentiate between Accept (aka Keep) Hunk vs File vs All Files? Same for Discard aka Undo?

**Comment by meganrogge** [comment]
Good question, I think we would. I will confirm with JooYoung today.

**Comment by AFre100** [comment]
@meganrogge yes, are we trying to get it into Insider's -- shipping, when 3/17?

**Comment by meganrogge** [comment]
1. Enable screen reader mode 
2. Generate some copilot edits
3. In the panel chat, hit `Keep` or `Undo` and verify that you hear sounds that indicate this has occurred
4. In the editor, hit `Keep` or `Undo` on individual changes and verify that you hear these sounds
